vm/wasmvm/exec/num.go

```diff
@@ -42,11 +42,23 @@ func (vm *VM) i32Popcnt() {
 }
 
 func (vm *VM) i32Add() {
-	vm.pushUint32(vm.popUint32() + vm.popUint32())
+
+	a := vm.popUint32()
+	b := vm.popUint32()
+
+	if IsI32AddOverflow(a, b) {
```

Comment:
this behave is against [wasm spec](https://webassembly.github.io/spec/core/bikeshed/index.html#op-iadd). Read the spec before implementation.